service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/AgentDirectoryMapperTest.java

```diff
@@ -1,82 +1,150 @@
+    private AgentPersonMapper agentPersonMapper;
+
     private AgentDirectoryMapper agentDirectoryMapper;
     private FhirParseService fhirParseService;
+    private MessageContext messageContext;
 
     @BeforeEach
     public void setUp() {
+        when(randomIdGeneratorService.createNewId()).thenReturn(TEST_ID);
+        messageContext = new MessageContext(randomIdGeneratorService);
+        lenient().when(agentPersonMapper.mapAgentPerson(any(), any())).thenAnswer(answerWithObjectId());
+
+        agentDirectoryMapper = new AgentDirectoryMapper(messageContext, agentPersonMapper);
         fhirParseService = new FhirParseService();
-        agentDirectoryMapper = new AgentDirectoryMapper(practitionerAgentPersonMapper, organizationToAgentMapper);
+    }
+
+    private Answer<String> answerWithObjectId() {
+        return invocation -> {
+            AgentDirectory.AgentKey agentKey = invocation.getArgument(0);
+            return String.format("<!--Mocked agentPerson for: %s %s -->",
+                agentKey.getPractitionerReference(),
+                agentKey.getOrganizationReference());
+        };
     }
 
     @Test
-    public void When_MappingAgentDirectory_Expect_CorrectAmountOfCallsToChildMappers() throws IOException {
-        setupMock(organizationToAgentMapper, practitionerAgentPersonMapper);
-        var jsonInput = ResourceTestFileUtils.getFileContent(BUNDLE_NO_ROLES);
-        var expectedOutput = ResourceTestFileUtils.getFileContent(DIRECTORY_NO_ROLES);
+    public void When_MappingAgentDirectory_Expect_CorrectOutputFromMapper() throws IOException {
+        var jsonInput = ResourceTestFileUtils.getFileContent(INPUT_AGENT_DIRECTORY);
         Bundle bundle = fhirParseService.parseResource(jsonInput, Bundle.class);
+        initializeMessageContextWithAgentKeys(bundle);
+        var expectedOutput = ResourceTestFileUtils.getFileContent(EXPECTED_AGENT_DIRECTORY);
+
         var mapperOutput = agentDirectoryMapper.mapEHRFolderToAgentDirectory(bundle, NHS_NUMBER);
+
         assertThat(mapperOutput).isEqualTo(expectedOutput);
     }
 
     @Test
-    public void When_MappingAgentDirectory_Expect_NoDuplicates() throws IOException {
-        setupMock(organizationToAgentMapper, practitionerAgentPersonMapper);
-        var jsonInput = ResourceTestFileUtils.getFileContent(BUNDLE_WITH_DUPLICATES);
-        var expectedOutput = ResourceTestFileUtils.getFileContent(DIRECTORY_NO_ROLES);
+    public void When_MappingAgentDirectoryWithoutPatientManagingOrganizationReference_Expect_Exception() throws IOException {
+        var jsonInput = ResourceTestFileUtils.getFileContent(INPUT_AGENT_DIRECTORY_WITHOUT_MANAGING_ORGANIZATION_REFERENCE);
+        Bundle bundle = fhirParseService.parseResource(jsonInput, Bundle.class);
+        initializeMessageContextWithAgentKeys(bundle);
+
+        assertThrows(
+            EhrMapperException.class,
+            () -> agentDirectoryMapper.mapEHRFolderToAgentDirectory(bundle, NHS_NUMBER)
+        );
```

Comment:
If we're using assertj elsewhere can we stick to assertj here too?

```suggestion
        assertThatThrownBy(() -> agentDirectoryMapper.mapEHRFolderToAgentDirectory(bundle, NHS_NUMBER))
            .isExactlyInstanceOf(EhrMapperException.class);
            // could also include .hasMessage("whatever we expect here")
```